crates/nu-command/src/filters/update_cells.rs

```diff
@@ -0,0 +1,265 @@
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        // the block to run on each cell
+        let engine_state = engine_state.clone();
+        let block: CaptureBlock = call.req(&engine_state, stack, 0)?;
+        let mut stack = stack.captures_to_stack(&block.captures);
+        let orig_env_vars = stack.env_vars.clone();
+        let orig_env_hidden = stack.env_hidden.clone();
+
+        let ctrlc = engine_state.ctrlc.clone();
+        let block: Block = engine_state.get_block(block.block_id).clone();
+
+        let span = call.head;
+
+        stack.with_env(&orig_env_vars, &orig_env_hidden);
+
+        // the columns to update
+        let columns: Option<Value> = call.get_flag(&engine_state, &mut stack, "columns")?;
+        let columns: Option<HashSet<String>> = match columns {
+            Some(val) => {
+                let cols = val
+                    .as_list()?
+                    .iter()
+                    .map(|val| val.as_string())
+                    .collect::<Result<Vec<String>, ShellError>>()?;
+                Some(HashSet::from_iter(cols.into_iter()))
+            }
+            None => None,
+        };
+
+        Ok(UpdateCellIterator {
+            input: input.into_iter(),
+            engine_state,
+            stack,
+            block,
+            columns,
+            span,
+        }
+        .into_pipeline_data(ctrlc))
+    }
+}
+
+struct UpdateCellIterator {
+    input: PipelineIterator,
+    columns: Option<HashSet<String>>,
+    engine_state: EngineState,
+    stack: Stack,
+    block: Block,
+    span: Span,
+}
+
+impl Iterator for UpdateCellIterator {
+    type Item = Value;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.input.next() {
+            Some(val) => {
+                if let Some(ref cols) = self.columns {
+                    if !val.columns().iter().any(|c| cols.contains(c)) {
+                        return Some(val);
+                    }
+                }
+
+                match val {
+                    Value::Record { vals, cols, span } => Some(Value::Record {
+                        vals: cols
+                            .iter()
+                            .zip(vals.into_iter())
+                            .map(|(col, val)| match &self.columns {
+                                Some(cols) if !cols.contains(col) => val,
+                                _ => process_cell(
+                                    val,
+                                    &self.engine_state,
+                                    &mut self.stack,
+                                    &self.block,
+                                    span,
+                                ),
+                            })
+                            .collect(),
+                        cols,
+                        span,
+                    }),
+                    // FIXME: is stepping into nested lists desired?
```

Comment:
I wouldn't step into nested lists. 

If the input is a list, each of its cells should be processed. If some of the cells are lists, they should be still treated as a single cell.